test/integration/e2eIntegrationTests.coffee

```diff
@@ -1027,4 +1027,106 @@ describe "e2e Integration Tests", ->
+      name: "TEST DATA - Mock endpoint 2"
+      urlPattern: "^/.*$"
+      priority: 2
+      allow: [ "PoC" ]
+      routes: [
+            name: "test route"
+            host: "localhost"
+            port: 1233
+            primary: true
+          ]
+    channel3 = new Channel
+      name: "TEST DATA - Mock endpoint 3"
+      urlPattern: "^/test/mock$"
+      priority: 1
+      allow: [ "PoC" ]
+      routes: [
+            name: "test route"
+            host: "localhost"
+            port: 1234
+            primary: true
+          ]
+
+    before (done) ->
+      config.authentication.enableMutualTLSAuthentication = false
+      config.authentication.enableBasicAuthentication = true
+
+      channel1.save (err) -> channel2.save (err) -> channel3.save (err) ->
+        testAppDoc =
+          clientID: "testApp"
+          clientDomain: "test-client.jembi.org"
+          name: "TEST Client"
+          roles:
+            [
+              "OpenMRS_PoC"
+              "PoC"
+            ]
+          passwordAlgorithm: "sha512"
+          passwordHash: "28dce3506eca8bb3d9d5a9390135236e8746f15ca2d8c86b8d8e653da954e9e3632bf9d85484ee6e9b28a3ada30eec89add42012b185bd9a4a36a07ce08ce2ea"
+          passwordSalt: "1234567890"
+          cert: ""
+
+        client = new Client testAppDoc
+        client.save (error, newAppDoc) ->
+          # Create mock endpoint to forward requests to
+          mockServer1 = testUtils.createMockServer 200, 'target1', 1233, ->
+            mockServer2 = testUtils.createMockServer 200, 'target2', 1234, done
+
+    after (done) ->
+      Channel.remove { name: "TEST DATA - Mock endpoint 1" }, ->
+        Channel.remove { name: "TEST DATA - Mock endpoint 2" }, ->
+          Channel.remove { name: "TEST DATA - Mock endpoint 3" }, ->
+            Client.remove { clientID: "testApp" }, ->
+              mockServer1.close ->
+                mockServer2.close ->
+                  done()
+
+    afterEach (done) ->
+      server.stop ->
+        done()
+
+    it "should route to the channel with higher priority if multiple channels match a request", (done) ->
+      server.start httpPort: 5001, ->
+        request("http://localhost:5001")
+          .get("/test/mock")
+          .auth("testApp", "password")
+          .expect(200)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              res.text.should.be.exactly 'target2' #should route to target2 via channel3
+              done()
+
+    it "should treat a channel with an undefined priority with lowest priority", (done) ->
+      server.start httpPort: 5001, ->
+        request("http://localhost:5001")
+          .get("/test/undefined/priority")
+          .auth("testApp", "password")
+          .expect(200)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              res.text.should.be.exactly 'target1' #should route to target1 via channel2
+              done()
```

Comment:
I simply though that it seem there is order implied by how the code is written so we should test for that.